client/src/App.js

```diff
@@ -111,7 +112,24 @@ export default function App() {
 
     history.push("/overview");
   }
+  // ================================ Darkmode
+  const [lightness, setLightness] = useState(
+    JSON.parse(localStorage.getItem("lightness")) || 0
+  );
+  function handleLightness(event) {
+    setLightness(event.target.value);
+  }
+  const setDarkestblack = useCallback(() => {
+    document.documentElement.style.setProperty(
+      "--lightness",
+      `${lightness / 8}%`
+    );
+    localStorage.setItem("lightness", JSON.stringify(lightness));
+  }, [lightness]);
 
+  useEffect(() => {
+    setDarkestblack();
+  }, [setDarkestblack, lightness]);

```

Comment:
No quite sure what happens here. Maybe you can explain it to me?

Reply:
I build my colors with HSL and defined the L part as a variable --lightness.

By default -lightness will be 0% which results in a true black. 
I created my other color values with the same pattern, and just added lightness increments of 15% steps for each lighter color.


Now in my code I can tweak the :root --lightness variable to change (line 123 - 126).
So if I change my base lightness from 0 to 40, 

my
 --color-black-100 will become 5% lighter, 
while my 
--color-black-40 will become 5% + 60% lighter. 



By default, my app should load in true black, unless somebody made a conscious choice to go with a lighter variant. Therefore I opted to write the preffered setting into localstorage and update the lightness setting as soon as the page mounts or something is being adjusted.